src/tree/mod.rs

```diff
@@ -499,7 +502,7 @@ impl RatchetTree {
                 .map(|&x| {
                     let pk = self.nodes[x.as_usize()].get_public_hpke_key().unwrap();
                     self.ciphersuite
-                        .hpke_seal(&pk, group_context, &[], &path_secret)
+                        .hpke_seal_secret(&pk, group_context, &[], &path_secret)
```

Comment:
Why is this a different function to `hpke_seal`?

Reply:
It's so we can seal Secrets directly without having to transform a secret into a slice first. But I agree that it's no necessary if we can't get rid of `encode` as well. The problem is that we're using `hpke_seal` not just on `Secret`s, but also, for example, on `GroupSecrets`, which contains `Secret`s as well. So one way to get rid of `encode` would be to have dedicated `hpke_seal` functions.